WIP: Fix tserver wait timeout

ddanielr opened the pull request:
Added a maximum number of retries to the tserver wait code. 
Fixed incorrect unit description in log messages.

ddanielr committed:
Fix tserver wait timeout

Added a maximum number of retries to the tserver wait code.
Fixed incorrect unit description in log messages.

EdColeman commented on server/master/src/main/java/org/apache/accumulo/master/Master.java:
```diff
@@ -1503,10 +1503,17 @@ private void blockForTservers() throws InterruptedException {
+    long timeIncrement = 15_000;
+    // maxWait is the maximum amount of time for retry commands to be attempted.
+    // Let's attempt the max amount of retries in the provided time duration.
+    long maxRetries = Math.round((double) (maxWait - initialWait) / (timeIncrement + logInterval));
```
I expect that log interval should not be part of the calculation,  Log interval is used to prevent spamming the logs with retry statements - say you have a 100ms timeout but want to wait 10s, you don't gain much from 10,000s of log staements saying retrying....  in that case, a log every few seconds so that you know its not dead is sufficient.

So, log interval does not have any impact on the wait time.

ddanielr committed:
Add basic wait loop

Switch to a wait loop based on time instead of using a retry object.

ddanielr committed:
Fix scan thread resource starvation (#3226)

This commit switches the filePermits semaphore to a "fair" semaphore
which ensures that threads requesting reserved readers are handled in a
FIFO queue vs ending up in a starvation state for resources.

ddanielr committed:
Merge remote-tracking branch 'upstream/1.10' into bugfix/retry-wait-logic
